tests/scripts/depends.py

```diff
@@ -500,15 +500,27 @@ def __init__(self, options, conf):
         # and padding modes are exercised separately) information by parsing
         # cipher.h, as the information is not readily available in mbedtls_config.h.
         cipher_info = CipherInfo()
-        # Find block cipher chaining and padding mode enabling macros by name.
-        cipher_chaining_symbols = self.config_symbols_matching(r'MBEDTLS_CIPHER_MODE_\w+\Z')
+
+        # Get block cipher chaining modes. Do not select ECB, it is always enabled.
```

Comment:
ECB is not always enabled. That was the case with the legacy API (which didn't really have ECB), but with the PSA API, ECB is a distinct mode that requires extra code and would typically be disabled in builds for constrained devices.

Reply:
So ECB needs to be tested.
The `CCM_STAR_NO_TAG` and `STREAM_CIPHER` is also defined by PSA and not by the legacy. Do they need to be tested here as well?

Reply:
Yes, eventually we should follow a similar principle: for each algorithm in a category, do a build with only that algorithm.

There may be intermediate stages where we exclude some algorithms because we need a lot of work to get them to pass the CI. But if so we should have an issue tracking the remaining work.